notifier/alertmanagerset.go

```diff
@@ -127,3 +132,62 @@ func (s *alertmanagerSet) configHash() (string, error) {
 	hash := md5.Sum(b)
 	return hex.EncodeToString(hash[:]), nil
 }
+
+func (s *alertmanagerSet) send(alerts ...*Alert) {
+	s.mtx.Lock()
+	defer s.mtx.Unlock()
+
+	if len(s.cfg.AlertRelabelConfigs) > 0 {
+		alerts = relabelAlerts(s.cfg.AlertRelabelConfigs, labels.Labels{}, alerts)
+		if len(alerts) == 0 {
+			return
+		}
+	}
+
+	for _, sendLoop := range s.sendLoops {
+		sendLoop.add(alerts...)
+	}
+}
+
+// addSendLoops creates and starts a send loop for newly discovered alertmanager.
+// This function expects the caller to acquire needed locks.
+func (s *alertmanagerSet) addSendLoops(ams []alertmanager) {
+	for _, am := range ams {
+		us := am.url().String()
+		// Only add if sendloop doesn't already exist
+		if loop, exists := s.sendLoops[us]; exists {
```

Comment:
this could be confusing, I'll try to avoid that as part of the follow up https://github.com/prometheus/prometheus/pull/16355/files#r2557623535

Reply:
There is a test with captures cases where we get duplicate sendloops after a reload.
Removing this logic makes that test fail.

Reply:
yep I saw that, we'll only need to start "new" loops. I'll look at that